dags/utils/run_result.py

```diff
@@ -0,0 +1,39 @@
+"""
+Copyright 2023 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License."""
+"""Utility functions for DAGs."""
+
+from dataclasses import dataclass, field
+from typing import Any, List, Mapping, Sequence
+
+MAX_TRIES = 3
+
+
+@dataclass
+class RunResult:
+  """Class for reporting the result of a DAG run."""
+
+  successful_hits: int = 0
+  failed_hits: int = 0
+  error_messages: Sequence[str] = field(default_factory=lambda: [])
+  retriable_events: Sequence[Mapping[str,
+                                     Any]] = field(default_factory=lambda: [])
+  dry_run: bool = False
+
+  def __add__(self, other: "RunResult") -> "RunResult":
```

Comment:
modify the add function with the new field to allow for `run_res1 + run_res2`. I assume that this should be causing a bug now, since RunResult constructor is being called with a problematic order here and [here](https://github.com/google-marketing-solutions/Tightlock/pull/126/files#diff-51e8701d15e07435b8e1c8d192569e7e8d14760ad95c69a3af5527ec1aba58c3R84)

Reply:
Done.